livebooks/create-a-swiftui-application.livemd

```diff
@@ -0,0 +1,290 @@
+
+<!-- livebook:{"break_markdown":true} -->
+
+You'll also need to manually navigate to the error tab (shown below) to trust and enable packages. Click on each error to trigger a prompt. Select **Trust & Enable All** to enable the plugin.
+
+The specific plugins are subject to change. At the time of writing you need to enable `LiveViewNativeStylesheetMacros`, `LiveViewNativeMacros`, and `CasePathMacros` as shown in the images below.
+
+<!-- livebook:{"break_markdown":true} -->
+
+![](https://github.com/liveview-native/documentation_assets/blob/main/trust-and-enable-liveview-native-stylesheet.png?raw=true)
+
+<!-- livebook:{"break_markdown":true} -->
+
+![](https://github.com/liveview-native/documentation_assets/blob/main/trust-and-enable-liveview-native-macros.png?raw=true)
+
+<!-- livebook:{"break_markdown":true} -->
+
+![](https://github.com/liveview-native/documentation_assets/blob/main/trust-and-enable-case-path-macros.png?raw=true)
+
+## Add the LiveView Live Form Package
+
+The [LiveView Native Live Form](https://github.com/liveview-native/liveview-native-live-form) provides forms for LiveView Native SwiftUI.
+
+To install LiveView Native Form, we need to add the [liveview-native-live-form](https://github.com/liveview-native/liveview-native-live-form) SwiftUI package to our iOS application. The steps will be mostly similar to what you have already setup with the `liveview-client-swiftui` package.
+
+Follow the [LiveView Native Form Installation Guide](https://github.com/liveview-native/liveview-native-live-form?tab=readme-ov-file#liveviewnativeliveform) on that project's README to add the `liveview-native-live-form` package to the SwiftUI application you created in the [Create a SwiftUI Application](https://hexdocs.pm/live_view_native_swiftui/create-a-swiftui-application.html) guide.
+
+Come back to this guide and continue after you have finished the installation process.
+
+## Setup the SwiftUI LiveView
+
+The [ContentView](https://developer.apple.com/tutorials/swiftui-concepts/exploring-the-structure-of-a-swiftui-app#Content-view) contains the main view of our iOS application.
+
+Replace the code in the `ContentView` file with the following to connect the SwiftUI application and the Phoenix application.
+
+<!-- livebook:{"break_markdown":true} -->
+
+```swift
+import SwiftUI
+import LiveViewNative
+
+struct ContentView: View {
+
+    var body: some View {
+        LiveView(.automatic(
+            development: .localhost(path: "/"),
+            production: .custom(URL(string: "https://example.com/")!)
+        ))
+    }
+}
+
+
+// Optionally preview the native UI in Xcode
+#Preview {
+    ContentView()
+}
+```
+
+<!-- livebook:{"break_markdown":true} -->
+
+The code above sets up the SwiftUI LiveView. By default, the SwiftUI LiveView connects to any Phoenix app running on http://localhost:4000.
```

Comment:
Can you give me an example of how that config would look? I don't have a physical iOS device to test that I've got it working correctly.

I'm assuming something like:

```swift
development: "http://192.XXX.XX.XXX/"
```

This feels like the instructions on finding an IP could muddy the guide. Maybe we should write a separate short guide on running the dev app on a physical device?

Reply:
It would be:

```swift
#LiveView(
  .automatic(
    development: URL(string: "http://192.168.1.xxx:4000")!,
    production: URL(string: "https://example.com")!
  )
)
```

Maybe the steps could be in a collapsible section like the example solutions are? You could add an Elixir cell that logs the IP address.

```elixir
IO.puts :os.cmd(~c[ipconfig getifaddr en0])
# or
{:ok, [_, {ip, _, _} | _]} = :inet.getif
IO.puts(ip |> Tuple.to_list |> Enum.map(&to_string/1) |> Enum.join("."))
```

Or hold off until we have that LVN Go app available so they can just scan the code to connect.

Reply:
I've added it in. Will push soon.